packages/shared/graphql/apolloClient.tsx

```diff
@@ -115,7 +116,13 @@ export function createApolloCache() {
 }
 
 function createHttpLink(token: string | undefined) {
-  const headers: Record<string, string> = {};
+  // We generate a client ID so that we can easily filter backend requests down to a specific client
```

Comment:
I don't know if this kind of comment is helpful or considered bad style. I think we could safely leave it out if people are likely to `git blame` to understand code, since my commit does a better job of explaining it anyways.

Reply:
TBH the thing I would want this comment to answer is "why isn't session id used?"

Reply:
That's fair! I have no idea what the async flow between apolloClient and session creation looks like, but if it's easy to wait for session creation here, I'd be totally fine with that.

Reply:
Ah, so this wasn't a security decision or anything, just a convenience one?

I recorded two Replays to compare the sign-in flow (because this is not something I'm super familiar with either)
* Authenticated: https://go/r/baac2125-c979-462d-af0a-8162468c97a2
* Unauthenticated: https://go/r/cd2eb8e5-2d95-4131-91a6-79ea8b31ea1d

tl;dr we initialize GraphQL/Apollo _before_ we have a session id, so maybe this light weight approach you have here is the right trade off.

Reply:
> we initialize GraphQL/Apollo before we have a session id

And we wait for the first GraphQL results (for recording and user metadata) before we even try to create a session.

Reply:
> Ah, so this wasn't a security decision or anything, just a convenience one?

I do think it has one relatively small security implication, which is that it would be possible for the user to change this value, and if on the graphql side we read this value as an authoritative answer as to the user's session we could potentially do something stupid like expose session details to an attacker. I think this is far-fetched and unlikely, but hey, those are usually the attack vectors that end up working! But it was mostly out of convenience.

> And we wait for the first GraphQL results (for recording and user metadata) before we even try to create a session.

Cool, so leaving it arbitrary is probably reasonable then 👍